Modules/_posixsubprocess.c

```diff
@@ -926,6 +927,11 @@ subprocess_fork_exec_impl(PyObject *module, PyObject *process_args,
     Py_ssize_t fds_to_keep_len = PyTuple_GET_SIZE(py_fds_to_keep);
 
     PyInterpreterState *interp = PyInterpreterState_Get();
+    if (interp->finalizing) {
+        PyErr_SetString(PyExc_RuntimeError,
+                        "can't create child process at interpreter shutdown");
```

Comment:
blocking all subprocess.Popen calls from working seems unnecessary. If anything add a `preexec_fn != Py_None &&` to this conditional as that is the only thing that could ever try to use the interpreter in the child process.